lib/Resque/Job/PID.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * PID tracker for the forked worker job.
+ *
+ * @package		Resque/Job
+ * @author		Chris Boulton <chris@bigcommerce.com>
+ * @license		http://www.opensource.org/licenses/mit-license.php
+ */
+class Resque_Job_PID
+{
+	/**
+	 * @var string The ID of the job this PID tracker class refers back to.
+	 */
+	private $id;
+
+	/**
+	 * Setup a new instance of the PID tracker class for the supplied job ID.
+	 *
+	 * @param string $id The ID of the job to track the PID for.
+	 */
+	public function __construct($id)
+	{
+		$this->id = $id;
+	}
+
+	/**
+	 * Create a new PID tracker item for the supplied job ID.
+	 *
+	 * @param string $id The ID of the job to track the PID of.
+	 */
+	public static function create($id)
+	{
+		Resque::redis()->set('job:' . $id . ':pid', (string)getmypid());
+	}
+
+	/**
+	 * Fetch the PID for the process actually executing the job.
+	 *
+	 * @return int PID of the process doing the job (on non-forking OS, PID of the worker, otherwise forked PID).
+	 */
+	public function get()
```

Comment:
Since this class doesn't let you do anything with the PID other than retrieve it, I'd just make this `static`, drop `__construct()` and `__toString()`, and pass the ID as an argument.